sympy/polys/densearith.py

```diff
@@ -1786,6 +1786,48 @@ def dmp_l1_norm(f, u, K):
     return sum([ dmp_l1_norm(c, v, K) for c in f ])
 
 
+def dup_l2_norm_squared(f, K):
+    """
+    Returns squared l2 norm of a polynomial in ``K[x]``.
+
+    Examples
+    ========
+
+    >>> from sympy.polys import ring, ZZ
+    >>> R, x = ring("x", ZZ)
+
+    >>> R.dup_l2_norm_squared(2*x**3 - 3*x**2 + 1)
+    14
+
+    """
+    if not f:
+        return K.zero
+    else:
+        return sum([ coeff**2 for coeff in f ])
```

Comment:
Maybe something like `return sum([coeff**2 for coeff in f], K.zero)` would suffice.

Reply:
I think so. I wrote it this way only to follow suit with the similar methods, like `dup_l1_norm`.